frontend/src/modules/entitydetails/components/OriginalStringProxy.js

```diff
@@ -0,0 +1,41 @@
+/* @flow */
+
+import * as React from 'react';
+
+import { FluentOriginalString } from 'modules/fluentoriginal';
+
+import OriginalString from './OriginalString';
+
+import type { Entity } from 'core/api';
+import type { Locale } from 'core/locales';
+
+
+type Props = {|
+    +entity: Entity,
+    +locale: Locale,
+    +pluralForm: number,
+    +handleClickOnPlaceable: (SyntheticMouseEvent<HTMLParagraphElement>) => void,
+|};
+
+
+/**
+ * Proxy for an OriginalString component based on the format of the entity.
+ *
+ * For Fluent strings ('ftl'), returns a Fluent-specific OriginalString
+ * component. For everything else, return the basic OriginalString component.
+ */
+export default function OriginalStringProxy(props: Props) {
+    if (props.entity.format === 'ftl') {
+        return <FluentOriginalString
+            entity={ props.entity }
+            handleClickOnPlaceable={ props.handleClickOnPlaceable }
+        />;
+    }
+
+    return <OriginalString
```

Comment:
I wonder if `GenericOriginalString` is a better name (for consistency with `*Editor` components)?

Reply:
Yes, I absolutely agree!